android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java

```diff
@@ -321,16 +331,26 @@ public void callJavascriptAfterLoad() {
       this.postDelayed(new Runnable() {
         @Override
         public void run() {
-          if(javascriptAfterLoad.size() > 0) {
-            loadUrl("javascript:" + javascriptAfterLoad.get(0));
-            javascriptAfterLoad.remove(0);
-            // Make sure we didn't reset the page in the middle of the queue!
-            if(keyboardSet) {
-              if (javascriptAfterLoad.size() > 0) {
-                callJavascriptAfterLoad();
-              }
+          StringBuilder allCalls = new StringBuilder();
+          if(javascriptAfterLoad.size() == 0) {
+            return;
+          }
+
+          while(javascriptAfterLoad.size() > 0) {
+            JSQueueEntry entry = javascriptAfterLoad.remove(0);
+            allCalls.append(entry.call);
+            allCalls.append(";");
+
+            if(entry.pause) {
+              break;
             }
           }
+
+          loadUrl("javascript:" + allCalls.toString());
```

Comment:
URL maximum length is 2048 characters. Just a wild one

Reply:
https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers